plugins/modules/win_find.py

```diff
@@ -37,6 +37,13 @@
         type: str
         choices: [ md5, sha1, sha256, sha384, sha512 ]
         default: sha1
+    depth:
+        description:
+            - Set the maximum number of levels to descend into.
+            - Setting recurse to C(false) will override this value, which is effectively depth 1.
+            - Default depth is effectively unlimited (2147483647).
+        type: int
+        default: 2147483647
```

Comment:
I spent far too much time dithering between `MaxValue` and `-1`. Went with `MaxValue` because it was fewer lines of code changed, but I agree `-1` has a better look & feel.  I'll push another commit with a `-1` implementation later today or tomorrow. Thanks @jborean93

Reply:
@jborean93 - instead of a `-1` default, would it be better or worse to mimic the standard [find](https://github.com/ansible/ansible/blob/79677c16f175e52a0dee9f2d366775e8ed0c8231/lib/ansible/modules/find.py#L124) module, which doesn't require a default?

Reply:
If it's possible sure, I just can't remember if the Ansible.Basic will coerce an unset integer value (`$null`) to `0` or not. Either one is fine with me really.

Reply:
Wouldn't this now cause an infinite recursion, `$val -ne 1` will always be true when you start at 0 because 0, -1, -2, -3, etc will never equal 1.

Reply:
Ah ok that makes more sense, it will still recurse to the actual nested folder structure.